packages/malloy-interfaces/thrift/malloy.thrift

```diff
@@ -277,10 +246,12 @@ struct Field {
   1: required Expression expression,
   // TODO only two kinds of distinguishable annotations are before `aggregate:` and before `name is value`
   // between `name` and `is`, or between `is` and `value` are converted to before `name`.
+  // TODO consider having an annotation tree with inheritance
   2: optional list<Annotation> annotations,
 }
 
 struct OrderBy {
+  // TODO consider making this a specialized reference like an "OrderByReference"
```

Comment:
Should the generic `Reference` (with name, path, and parameters) be used a) everywhere b) most places, with some exceptions like `OrderByReference` c) never; all should be specialized.

Reply:
Currently, everything uses `Reference` even if a) they only allow a name and not a path, or b) they don't support parameters.

Reply:
Michael's vote (not super strong vote) is that there is a separate reference type.